integrated_channels/integrated_channel/course_metadata.py

```diff
@@ -0,0 +1,127 @@
+    course_run_details = course_run_details.copy()
+    for key, base_course_value in course_details.items():
+        if course_run_details.get(key) is None and key not in EXCLUDED_COURSE_DETAIL_KEYS:
+            course_run_details[key] = base_course_value
+    return course_run_details
+
+
+def get_course_list(user, enterprise_customer):
+    """
+    Determine which catalog belongs to a given Enterprise Customer and list the course runs it contains.
+
+    Args:
+        user: A Django user requesting the course list
+        enterprise_customer: The given Enterprise Customer
+
+    Returns:
+        iterable: An iterable containing the details of each course run.
+    """
+    catalog_id = enterprise_customer.catalog
+
+    if catalog_id is None:
+        LOGGER.info('No catalog attached to %s', enterprise_customer)
+    else:
+        client = CourseCatalogApiClient(user)
+
+        catalog_courses = client.get_single_catalog_courses(catalog_id)
+        LOGGER.info('Retrieving course list for catalog %s', catalog_id)
+
+        for course in catalog_courses:
+            course_key = course.get('key')
+            course_details = client.get_single_course_details(course_key)
+            for run in course_details.get('course_runs', []):
+                yield updated_course_run(course_details, run)
+
+
+class BaseCourseExporter(object):
+    """
+    Base class for course metadata exporters to implement a "send" method on.
+    """
+
+    data_transform = {}
+
+    def __init__(self, user, plugin_configuration):
+        """
+        Save the appropriate details for use elsewhere in the object.
+        """
+        self.user = user
+        self.enterprise_customer = plugin_configuration.enterprise_customer
+        self.plugin_configuration = plugin_configuration
+
+    def get_parsed(self):
+        """
+        Retrieve the course runs for the linked catalog, and parse them appropriately.
+        """
+        results = (self.process_course_run(run) for run in get_course_list(self.user, self.enterprise_customer))
+        return results
+
+    def process_course_run(self, course_run_details):
+        """
+        Parse the provided course into the format natively supported by the provider.
+        """
+        LOGGER.info(
+            'Processing course with ID %s',
+            course_run_details['key']
+        )
+        LOGGER.debug(
+            'Parsing course for %s: %s',
+            self.enterprise_customer,
+            json.dumps(course_run_details, indent=4),
+        )
+        output = {}
+        for key, transform in self.data_transform.items():
+            output[key] = transform(course_run_details) if transform is not None else course_run_details.get(key)
+        return output
+
+    def send_all(self):
+        """
+        Wrapper for sending all course metadata blocks.
+        """
+        for course in self.get_parsed():
+            self.send(course)
+
+    def send(self, course):
```

Comment:
As with the other function names it would be great if we could be more specific about what is going on -- eg `transmit_course_metadata`